homeassistant/components/onewire/sensor.py

```diff
@@ -30,33 +39,137 @@
     "28": {"temperature": "temperature"},
     "3B": {"temperature": "temperature"},
     "42": {"temperature": "temperature"},
+    "1D": {"counter_a": "counter.A", "counter_b": "counter.B"},
+    "EF": {"HobbyBoard": "special"},
+}
+
+# EF sensors are usually hobbyboards specialized sensors.
+# These can only be read by OWFS.  Currently this driver only supports them
+# via owserver (network protocol)
+
+HOBBYBOARD_EF = {
+    "HobbyBoards_EF": {
+        "humidity": "humidity/humidity_corrected",
+        "humidity_raw": "humidity/humidity_raw",
+        "temperature": "humidity/temperature",
+    },
+    "HB_MOISTURE_METER": {
+        "moisture_0": "moisture/sensor.0",
+        "moisture_1": "moisture/sensor.1",
+        "moisture_2": "moisture/sensor.2",
+        "moisture_3": "moisture/sensor.3",
+    },
+    "HB_HUB": {
+        "branch_0": "hub/branch.0",
+        "branch_1": "hub/branch.1",
+        "branch_2": "hub/branch.2",
+        "branch_3": "hub/branch.3",
+        "short_0": "hub/short.0",
+        "short_1": "hub/short.1",
+        "short_2": "hub/short.2",
+        "short_3": "hub/short.3",
+    },
 }
 
 SENSOR_TYPES = {
     "temperature": ["temperature", TEMP_CELSIUS],
     "humidity": ["humidity", "%"],
+    "humidity_raw": ["humidity", "%"],
     "pressure": ["pressure", "mb"],
     "illuminance": ["illuminance", "lux"],
+    "wetness_0": ["wetness", "%"],
+    "wetness_1": ["wetness", "%"],
+    "wetness_2": ["wetness", "%"],
+    "wetness_3": ["wetness", "%"],
+    "moisture_0": ["moisture", "cb"],
+    "moisture_1": ["moisture", "cb"],
+    "moisture_2": ["moisture", "cb"],
+    "moisture_3": ["moisture", "cb"],
+    "counter_a": ["counter", "count"],
+    "counter_b": ["counter", "count"],
+    "HobbyBoard": ["none", "none"],
+    "branch_0": ["presense", "present"],
+    "branch_1": ["presense", "present"],
+    "branch_2": ["presense", "present"],
+    "branch_3": ["presense", "present"],
+    "short_0": ["problem", "short"],
+    "short_1": ["problem", "short"],
+    "short_2": ["problem", "short"],
+    "short_3": ["problem", "short"],
 }
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_NAMES): {cv.string: cv.string},
         vol.Optional(CONF_MOUNT_DIR, default=DEFAULT_MOUNT_DIR): cv.string,
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=4304): cv.port,
     }
 )
 
 
+def hb_info_from_type(type="std"):
+    """Return the proper info array for the device type."""
+    if "std" in type:
+        return DEVICE_SENSORS
+    if "HobbyBoard" in type:
+        return HOBBYBOARD_EF
+
+
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the one wire Sensors."""
     base_dir = config.get(CONF_MOUNT_DIR)
+    owport = config.get(CONF_PORT)
+    owhost = config.get(CONF_HOST)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.

Reply:
Question about this one.  I'm looking at components/sigfox as an example.  So I should set it to  owport = config[CONF_PORT] I believe.

Then when do you use the config.get(X) function, or is the answer never?  Should the CONF_MOUNT_DIR one be changed too?

Reply:
If the item is optional without default we use `dict.get`.